schema/src/serialize.rs

```diff
@@ -0,0 +1,317 @@
+        }
+
+        seq.end()
+    }
+}
+impl PartialEq for WithContext<'_, '_, NamedFieldsMeta> {
+    fn eq(&self, other: &Self) -> bool {
+        if self.data.declarations.len() != other.data.declarations.len() {
+            return false;
+        }
+
+        self.data
+            .declarations
+            .iter()
+            .zip(other.data.declarations.iter())
+            .all(|(self_declaration, other_declaration)| {
+                self_declaration.name == other_declaration.name
+                    && self.type_name(self_declaration.ty) == other.type_name(other_declaration.ty)
+            })
+    }
+}
+
+impl Serialize for WithContext<'_, '_, UnnamedFieldsMeta> {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let mut seq = serializer.serialize_tuple(self.data.types.len())?;
+
+        for &type_id in &self.data.types {
+            seq.serialize_element(self.type_name(type_id))?;
+        }
+
+        seq.end()
+    }
+}
+impl PartialEq for WithContext<'_, '_, UnnamedFieldsMeta> {
+    fn eq(&self, other: &Self) -> bool {
+        if self.data.types.len() != other.data.types.len() {
+            return false;
+        }
+
+        self.data
+            .types
+            .iter()
+            .zip(other.data.types.iter())
+            .all(|(&self_type, &other_type)| {
+                self.type_name(self_type) == other.type_name(other_type)
+            })
+    }
+}
+
+impl Serialize for WithContext<'_, '_, EnumMeta> {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let mut seq = serializer.serialize_seq(Some(self.data.variants.len()))?;
+
+        for variant in &self.data.variants {
+            seq.serialize_element(&variant.add_ctx(self.context))?
+        }
+
+        seq.end()
+    }
+}
+impl PartialEq for WithContext<'_, '_, EnumMeta> {
+    fn eq(&self, other: &Self) -> bool {
+        self.data
+            .variants
+            .iter()
+            .zip(other.data.variants.iter())
+            .all(|(self_variant, other_variant)| {
+                self_variant.add_ctx(self.context) == other_variant.add_ctx(other.context)
+            })
+    }
+}
+
+impl Serialize for WithContext<'_, '_, EnumVariant> {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        if let Some(type_id) = self.data.ty {
+            let mut map = serializer.serialize_map(Some(3))?;
+            map.serialize_entry("name", &self.data.tag)?;
+            map.serialize_entry("type", self.type_name(type_id))?;
+            map.serialize_entry("discriminant", &self.data.discriminant)?;
+            map.end()
+        } else {
+            let mut map = serializer.serialize_map(Some(2))?;
+            map.serialize_entry("name", &self.data.tag)?;
+            map.serialize_entry("discriminant", &self.data.discriminant)?;
+            map.end()
+        }
```

Comment:
```suggestion
        if let Some(type_id) = self.data.ty {
            let mut map = serializer.serialize_map(Some(3))?;
            map.serialize_entry("name", &self.data.tag)?;
            map.serialize_entry("type", self.type_name(type_id))?;
            map.serialize_entry("dis", &self.data.discriminant)?;
            map.end()
        } else {
            let mut map = serializer.serialize_map(Some(2))?;
            map.serialize_entry("name", &self.data.tag)?;
            map.serialize_entry("dis", &self.data.discriminant)?;
            map.end()
        }
```

Reply:
I disagree here, we don't use abbreviations much. Is there some benefit?

Reply:
removed `discriminant` as it's not used